android/app/src/main/kotlin/com/chicagoroboto/features/main/MainActivity.kt

```diff
@@ -69,30 +68,37 @@ class MainActivity : AppCompatActivity(), SessionNavigator, SpeakerNavigator, Na
 
     private fun showView(viewId: Int): Boolean {
         val view: View? = when (viewId) {
-            R.id.action_schedule -> SessionDateView(this)
+            R.id.action_schedule -> SessionDateView(this).setupTabLayout(tabs)
```

Comment:
This one also involves intimate knowledge of children.  Perhaps, when the view is attached, it can check if it's activity implements something like a `TabHolder` interface and, if so, get and configure the tabs from there (and disable them on detach).

Reply:
I created a `TabHolder` interface for the Activity to implement that will return a `TabLayout?`, so the appropriate child views can check and see if it's parent is of type `TabHolder`, and if so, get a reference to the tab layout.